src/subscriber.ts

```diff
@@ -184,6 +211,25 @@ export class Message {
     }
   }
 
+  /**
+   * Acknowledges the message, expecting a response (for exactly once subscriptions).
+   *
+   * @example
+   * ```
+   * subscription.on('message', async (message) => {
+   *   const response = await message.ackWithResponse();
+   * });
+   * ```
+   */
+  async ackWithResponse(): Promise<AckResponse> {
+    if (!this._handled) {
```

Comment:
Can we short-circuit this in case exactly once delivery is disabled and return success?

Reply:
I had to think about this, but I think you're right, it's actually returning the incorrect response for non-EOS cases.